packages/flutter/lib/src/widgets/reorderable_list.dart

```diff
@@ -974,9 +963,9 @@ class SliverReorderableListState extends State<SliverReorderableList> with Ticke
     return Rect.fromLTWH(origin.dx, origin.dy, _dragInfo!.itemSize.width, _dragInfo!.itemSize.height);
   }
 
-  Offset _itemOffsetAt(int index) {
-    final RenderBox itemRenderBox =  _items[index]!.context.findRenderObject()! as RenderBox;
-    return itemRenderBox.localToGlobal(Offset.zero);
+  Offset _itemOffsetAt(int index, {bool extent = false}) {
```

Comment:
What does `extent` here represent?

Reply:
The `extent` parameter in the `_itemOffsetAt` function determines which corner of the item's geometry we are interested in. If `extent` is `true`, we return the 'end' corner of the item - the bottom right corner for vertical scrolling and the top right corner for horizontal scrolling. If `extent` is `false`, we return the 'start' corner of the item - the top left corner for both vertical and horizontal scrolling. This allows us to calculate the offset of the item in the list based on its position and size.

Reply:
Thank you for explaining! Maybe we should just pass in the correct geometry rather than a bool. I think that would be easier for maintainers to follow.